vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -76,15 +76,15 @@ public AbstractGridMultiSelectionModel(Grid<T> grid) {
 
         selectionColumn
                 .setSelectAllCheckBoxVisibility(isSelectAllCheckboxVisible());
-        grid.getElement().getNode().runWhenAttached(ui -> grid.getElement()
```

Comment:
Why is the deferring removed? I'm sure there was a purpose it was added there originally.

Reply:
This idea is that 
1. There is a button with addEventListener, the code  `grid.getElement().getNode().runWhenAttached(ui -> grid.getElement()...` runs immediately because the `grid.getElement()` is attached already. So the current selection mode (`grid.getSelectionModel`) is still SINGLE => the condition  `grid.getSelectionModel == this` is never true, test case: trigger change Selection Mode to MULTI by clicking button is failed. 
2. If we do not have the checking condition `grid.getSelectionModel == this`, like the code of master branch, when we change Selection mode to others like NONE/SINGLE from MULTI, there is a bug, because ` grid.getElement()
                        .insertChild(0, selectionColumn.getElement());` runs after remove method.

So my old idea is to...remove `runWhenAttached ` and the grid can insert the column checkbox.

But I changed code with a new fixing version.